### DISCUSSION TITLE
reset counter after power off

### DISCUSSION THREAD

**Discussion by maury77** [arendst/Tasmota#13152] [Q&A] [answered]
I have a problem with a counter connected a water pulse sensor.

When I turn off the power to the device,  the counter resets to zero value.

Is there a flag that allows to persist value after power fail

**Comment by barbudor** [comment, marked as answer]
Saving the counter value from power loss would require to write the value continuously to the flash which will wear off the chip in a matter of months.
I believe this is a DIY system, so you can easily add a small LiPo battery to gain some hours of safety.
You can also handle that in the backend system, detecting that the device has been restarted and the counter reset.
